### PR TITLE
manifest: Properly change the app id in the appstream file

### LINKED ISSUE #90: Renaming an applications id should add provides/id to appstream file
If you rename foo.desktop and foo.appdata.xml to org.bar.Foo then it should label such in the appdata file so software can better show this.

```xml
<provides>
  <id>foo.desktop</id>
</provides>
```

### PR DESCRIPTION
**Pull request by hfiguiere**
It include more testing

Close #90

### CODE REVIEW

File: src/builder-manifest.c [hunk 1]
```diff
@@ -2827,13 +2832,27 @@ builder_manifest_cleanup (BuilderManifest *self,
...
                     }
                 }
 
+              n_provides = flatpak_xml_find (n_root, "provides", NULL);
+              if (!n_provides)
+                {
+                  n_provides = flatpak_xml_new ("provides");
+                  flatpak_xml_add (n_root, n_provides);
+                }
+              if (n_provides)
```
Comment (TingPing) [inline src/builder-manifest.c:2848]: Will this always be true?

Comment (hfiguiere) [reply src/builder-manifest.c:2848]: likely. But I try to avoid null derefences.

Comment (hfiguiere) [reply src/builder-manifest.c:2848]: but then the same checks aren't done elsewhere.

Comment (TingPing) [reply src/builder-manifest.c:2848]: `flatpak_xml_new` can't fail and always returns something.

**Pull request merged**